### ISSUE TITLE
NOTICE and USERNOTICE messages loaded from recent messages service have incorrect timestamp

### ISSUE THREAD

**Issue by RAnders00** [Chatterino/chatterino2#1286]
**Describe the bug**
When you load a channel where `NOTICE`s (or `USERNOTICE`s) are in the last 500 messages (e.g. a sub-mode-on message), then the timestamp on those messages have the wrong timestamp.

**To Reproduce**
- find a channel with a message like `This room is now in emote-only mode.`, where that message is at least 1 minute old (so you can notice the difference)
- reopen chatterino
- observe that the timestamp on the message is the time of opening chatterino

**Expected behavior**
The timestamp is calculated from the `rm-received-ts` tag, and not the current time. E.g.:

```
@rm-received-ts=1567706276587;historical=1;msg-id=emote_only_on :tmi.twitch.tv NOTICE #trihex :This room is now in emote-only mode.
```

is a message that the `recent-messages` service returns. `rm-received-ts` is **always** set.

**Screenshots**

![image](https://user-images.githubusercontent.com/1629196/64387273-be8bed80-d03b-11e9-8b6e-18aad814e28f.png)

**Chatterino version**
2.1.2

**Additional context**

**Comment by savabozo** [comment]
Might be related to #1269

**Comment by fourtf** [comment]
is this still relevant?

**Comment by RAnders00** [comment]
Yes, still occurs, also affects CLEARCHAT I believe

**Comment by dnsge** [comment]
This was fixed for `NOTICE`'s in f41b026 [here](https://github.com/dnsge/chatterino2/blob/f41b026a0fae427880e1378d71a5cdd8da541019/src/providers/twitch/IrcMessageHandler.cpp#L637) but still needs to be fixed for `USERNOTICE`'s and whatever else. The code can probably be copied with minimal changes.
